Detectors/TPC/qc/src/Clusters.cxx

```diff
@@ -0,0 +1,95 @@
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "TH1.h"
+#include "TH2.h"
+#include "TFile.h"
+
+#include "TPCQC/Clusters.h"
+#include "TPCBase/Painter.h"
+#include "TPCBase/ROC.h"
+#include "TPCBase/Mapper.h"
+
+ClassImp(o2::tpc::qc::Clusters);
+
+using namespace o2::tpc::qc;
+
+//______________________________________________________________________________
+bool Clusters::processCluster(o2::tpc::ClusterNative const& cluster, o2::tpc::Sector const sector, const int row)
+{
+  const int nROC = row < 63 ? int(sector) : int(sector) + 36;
+  const int rocRow = row < 63 ? row : row - 63;
+
+  const float pad = cluster.getPad();
+
+  const uint16_t qMax = cluster.qMax;
+  const uint16_t qTot = cluster.qTot;
+  const float sigmaPad = cluster.getSigmaPad();
+  const float sigmaTime = cluster.getSigmaTime();
+  const float timeBin = cluster.getTime();
+
+  float count = mNClusters.getCalArray(nROC).getValue(rocRow, pad);
+  mNClusters.getCalArray(nROC).setValue(rocRow, pad, count + 1);
+
+  float charge = mQMax.getCalArray(nROC).getValue(rocRow, pad);
+  mQMax.getCalArray(nROC).setValue(rocRow, pad, charge + qMax);
+
+  charge = mQTot.getCalArray(nROC).getValue(rocRow, pad);
+  mQTot.getCalArray(nROC).setValue(rocRow, pad, charge + qTot);
+
+  count = mSigmaTime.getCalArray(nROC).getValue(rocRow, pad);
+  mSigmaTime.getCalArray(nROC).setValue(rocRow, pad, count + sigmaTime);
+
+  count = mSigmaPad.getCalArray(nROC).getValue(rocRow, pad);
+  mSigmaPad.getCalArray(nROC).setValue(rocRow, pad, count + sigmaPad);
+
+  count = mTimeBin.getCalArray(nROC).getValue(rocRow, pad);
+  mTimeBin.getCalArray(nROC).setValue(rocRow, pad, count + timeBin);
+
+  return true;
+}
+
+//______________________________________________________________________________
+void Clusters::analyse()
+{
+  mQMax /= mNClusters;
+  mQTot /= mNClusters;
+  mSigmaTime /= mNClusters;
+  mSigmaPad /= mNClusters;
+  mTimeBin /= mNClusters;
+}
+
+//______________________________________________________________________________
+void Clusters::dumpToFile(std::string filename)
+{
+  if (filename.find(".root") != std::string::npos) {
+    filename.resize(filename.size() - 5);
+  }
+
+  std::string canvasFile = filename + "_canvas.root";
+  auto f = std::unique_ptr<TFile>(TFile::Open(canvasFile.c_str(), "recreate"));
+  f->WriteObject(o2::tpc::painter::draw(mNClusters), "N_Clusters");
+  f->WriteObject(o2::tpc::painter::draw(mQMax), "Q_Max");
+  f->WriteObject(o2::tpc::painter::draw(mQTot), "Q_Tot");
+  f->WriteObject(o2::tpc::painter::draw(mSigmaTime), "Sigma_Time");
+  f->WriteObject(o2::tpc::painter::draw(mSigmaPad), "Sigma_Pad");
+  f->WriteObject(o2::tpc::painter::draw(mTimeBin), "Time_Bin");
+  f->Close();
```

Comment:
Why not use the name of the object? E.g.
```c++
f->WriteObject(o2::tpc::painter::draw(mTimeBin), mTimeBin.getName());
```
and so on. Same below.

Reply:
done